### DISCUSSION TITLE
How can I search in document library from the root folder with fields that exist in the files to get the target files using pnp modern search

### DISCUSSION THREAD

**Discussion by adnanjalloh** [microsoft-search/pnp-modern-search#2187] [Q&A]
Hi, This is not a bug report it's just a question,

if I am in the root folder in a document library and want to search for the files with fields that don't exist in the root folder,

is there a way to do that with PNP modern search?

thanks

**Comment by wobba** [comment]
Take a look at https://docs.microsoft.com/en-us/microsoft-365/community/how-do-site-columns-become-managed-properties-thus-available-for-search to get you started on SharePoint column to crawled property to managed property.
